bundles/binding/org.openhab.binding.autelis/src/main/java/org/openhab/binding/autelis/internal/AutelisBinding.java

```diff
@@ -0,0 +1,357 @@
+
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import org.openhab.binding.autelis.AutelisBindingProvider;
+import org.apache.commons.lang.StringUtils;
+import org.openhab.core.binding.AbstractActiveBinding;
+import org.openhab.core.items.Item;
+import org.openhab.core.library.items.NumberItem;
+import org.openhab.core.library.items.SwitchItem;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.IncreaseDecreaseType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.io.net.http.HttpUtil;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.InputSource;
+	
+
+/**
+ * Autelis Pool Control Binding
+ * 
+ * Autelis controllers allow remote access to many common pool systems. This
+ * binding allows openHAB to both monitor and control a pool system through these 
+ * controllers.
+ * 
+ * @see <a href="http://Autelis.com">http://autelis.com</a>
+ * 
+ * @author Dan Cunningham
+ * @since 1.7.0
+ */
+public class AutelisBinding extends AbstractActiveBinding<AutelisBindingProvider> {
+
+	private static final Logger logger = 
+		LoggerFactory.getLogger(AutelisBinding.class);
+
+	/**
+	 * Default port to use for connections to a Autelis controller
+	 */
+	static final int DEFAULT_PORT=80;
+	
+	/**
+	 * Default timeout for http connections to a Autelis controller
+	 */
+	static final int TIMEOUT=5000;
+	
+	/**
+	 * Configurable Host or ip address for http connections to a Autelis controller
+	 */
+	private String host;
+	
+	/**
+	 * Configurable port for http connections to a Autelis controller
+	 */
+	private int port;
+	
+	/**
+	 * Configurable username http connections to a Autelis controller
+	 */
+	private String username;
+	
+	/**
+	 * Configurable password http connections to a Autelis controller
+	 */
+	private String password;
+	
+	/**
+	 * Constructed URL consisting of host, port, username and password use to connect to a Autelis controller
+	 */
+	private String baseURL;
+	
+	/**
+	 * Regex expression to match XML responses from the Autelis, this is used to combine similar XML docs
+	 * into a single document, {@link XPath} is still used for XML querying
+	 */
+	private Pattern responsePattern = Pattern.compile("<response>(.+?)</response>", Pattern.DOTALL);
+	
+	/**
+	 * Commands can only be sent on equipment and temp configurations
+	 */
+	private Pattern commandPattern = Pattern.compile("^(equipment|temp)\\.(.*)");
+	
+	/**
+	 * We cache previous values for {@link Item}s using the item name as a key.  After a polling
+	 * run has been executed we only update an item if the value is different then what's in the
+	 * cache. Not sure if there's a better way to do this.
+	 */
+	private Map<String, State> cachedItemState;
```

Comment:
In general Bindings should be stateless since the Items' state can (potentially) be changed from other instances as well. Hence such caching should be avoided. See also this recent discussion about the usage of the ItemRegistry directly https://groups.google.com/d/msg/openhab/UeexB3MRLBU/HmmxrR4WUvoJ.